conan/tools/cmake/cmake.py

```diff
@@ -66,13 +66,13 @@ def configure(self, source_folder=None):
         if not self._conanfile.should_configure:
             return
 
-        source = self._conanfile.source_folder
+        source = self._conanfile.layout.source_folder
```

Comment:
Wasn't the initial idea of the PR0 that this is not necessary? Most Conan usage can still be done via ``self._conanfile.source_folder``?

Reply:
> Wasn't the initial idea of the PR0 that this is not necessary?

No, the idea was to not break, but I thought that we will promote the usage of the `layout` instead of the conanfile directly. That's why the "getters" in this PR have the deprecated function.